format of parameters leads to bad generated code

If a parameter has a format the generated code will not compile:
Examples: 
```json
{
  "consumes": [
    "application/json",
    "application/x-www-form-urlencoded"
  ],
  "paths": {
    "/testing": {
      "put": {
        "parameters": [
          {
            "format": "date",
            "in": "formData",
            "name": "testingThis",
            "type": "string"
          }
        ],
        "responses": {
          "200": {
            "description": "Success message"
          }
        }
      }
    }
  },
  "produces": [
    "application/json",
    "application/json; charset=utf-8"
  ],
  "swagger": "2.0"
}
```

And some console output:

``` shell
$ swagger generate client -f testapi.json
2016/01/06 13:20:16 rendered client parameters template: operations.PutTestingParameters
2016/01/06 13:20:16 rendered client responses template: operations.PutTestingResponses
2016/01/06 13:20:16 rendered operation group client template: operations.OperationsClient
2016/01/06 13:20:16 rendered client embedded swagger JSON template: client.SwaggerClient
2016/01/06 13:20:16 rendered client facade template: client.SwaggerClient
$ cd client/
$ go build
operations/put_testing_parameters.go:41: invalid operation: fTestingThis != "" (mismatched types strfmt.Date and string)
operations/put_testing_parameters.go:42: cannot use fTestingThis (type strfmt.Date) as type string in argument to r.SetFormParam
```

This was found with a parameter of an array type with elements which had format: 

```json
{
  "consumes": [
    "application/json",
    "application/x-www-form-urlencoded"
  ],
  "paths": {
    "/testing": {
      "put": {
        "parameters": [
          {
            "in": "formData",
            "items": {
              "format": "date",
              "type": "string"
            },
            "name": "testingThis",
            "type": "array"
          }
        ],
        "responses": {
          "200": {
            "description": "Success message"
          }
        }
      }
    }
  },
  "produces": [
    "application/json",
    "application/json; charset=utf-8"
  ],
  "swagger": "2.0"
}
```
Which does not compile with a similar error:
```shell
$ swagger generate client -f testapiarray.json
2016/01/06 13:32:05 rendered client parameters template: operations.PutTestingParameters
2016/01/06 13:32:05 rendered client responses template: operations.PutTestingResponses
2016/01/06 13:32:05 rendered operation group client template: operations.OperationsClient
2016/01/06 13:32:05 rendered client embedded swagger JSON template: client.SwaggerClient
2016/01/06 13:32:06 rendered client facade template: client.SwaggerClient
$ cd client/
$ go build
operations/put_testing_parameters.go:41: cannot use valuesTestingThis (type []strfmt.Date) as type []string in argument to swag.JoinByFormat
```

And final example:
If the type of the items is referenced:
```json
{
  "consumes": [
    "application/json",
    "application/x-www-form-urlencoded"
  ],
  "definitions": {
    "testType": {
      "format": "date",
      "type": "string"
    }
  },
  "paths": {
    "/testing": {
      "put": {
        "parameters": [
          {
            "in": "formData",
            "items": {
              "$ref": "#/definitions/testType"
            },
            "name": "testingThis",
            "type": "array"
          }
        ],
        "responses": {
          "200": {
            "description": "Success message"
          }
        }
      }
    }
  },
  "produces": [
    "application/json",
    "application/json; charset=utf-8"
  ],
  "swagger": "2.0"
}
```
Which results in 
```shell
$ swagger generate client -f testapi2.json
2016/01/06 13:38:37 rendered model template: testType
2016/01/06 13:38:37 generated model testType
2016/01/06 13:38:37 rendered client parameters template: operations.PutTestingParameters
2016/01/06 13:38:37 put_testing_parameters.go:32:17: expected type, found newline (and 5 more errors)
2016/01/06 13:38:37 rendered client responses template: operations.PutTestingResponses
2016/01/06 13:38:37 rendered operation group client template: operations.OperationsClient
2016/01/06 13:38:37 rendered client embedded swagger JSON template: client.SwaggerClient
2016/01/06 13:38:37 rendered client facade template: client.SwaggerClient
$ cd client/
$ go build
operations/put_testing_parameters.go:32: syntax error: unexpected semicolon or newline
operations/put_testing_parameters.go:38: syntax error: unexpected )
operations/put_testing_parameters.go:40: non-declaration statement outside function body
operations/put_testing_parameters.go:41: syntax error: unexpected }
```

--- a/strfmt/default.go
+++ b/strfmt/default.go
@@ -187,6 +187,10 @@ func (b Base64) Value() (driver.Value, error) {
 	return driver.Value(string(b)), nil
 }
 
+func (b Base64) String() string {
+	return string(b)
+}
+
 // URI represents the uri string format as specified by the json schema spec
 //
 // swagger:strfmt uri
@@ -220,6 +224,10 @@ func (u *URI) Scan(raw interface{}) error {
 // Value converts a value to a database driver value
 func (u URI) Value() (driver.Value, error) {
 	return driver.Value(string(u)), nil
+}
+
+func (u URI) String() string {
+	return string(u)
 }
 
 // Email represents the email string format as specified by the json schema spec
@@ -257,6 +265,10 @@ func (e Email) Value() (driver.Value, error) {
 	return driver.Value(string(e)), nil
 }
 
+func (e Email) String() string {
+	return string(e)
+}
+
 // Hostname represents the hostname string format as specified by the json schema spec
 //
 // swagger:strfmt hostname
@@ -290,6 +302,10 @@ func (h *Hostname) Scan(raw interface{}) error {
 // Value converts a value to a database driver value
 func (h Hostname) Value() (driver.Value, error) {
 	return driver.Value(string(h)), nil
+}
+
+func (h Hostname) String() string {
+	return string(h)
 }
 
 // IPv4 represents an IP v4 address
@@ -327,6 +343,10 @@ func (u IPv4) Value() (driver.Value, error) {
 	return driver.Value(string(u)), nil
 }
 
+func (u IPv4) String() string {
+	return string(u)
+}
+
 // IPv6 represents an IP v6 address
 //
 // swagger:strfmt ipv6
@@ -360,6 +380,10 @@ func (u *IPv6) Scan(raw interface{}) error {
 // Value converts a value to a database driver value
 func (u IPv6) Value() (driver.Value, error) {
 	return driver.Value(string(u)), nil
+}
+
+func (u IPv6) String() string {
+	return string(u)
 }
 
 // UUID represents a uuid string format
@@ -397,6 +421,10 @@ func (u UUID) Value() (driver.Value, error) {
 	return driver.Value(string(u)), nil
 }
 
+func (u UUID) String() string {
+	return string(u)
+}
+
 // UUID3 represents a uuid3 string format
 //
 // swagger:strfmt uuid3
@@ -430,6 +458,10 @@ func (u *UUID3) Scan(raw interface{}) error {
 // Value converts a value to a database driver value
 func (u UUID3) Value() (driver.Value, error) {
 	return driver.Value(string(u)), nil
+}
+
+func (u UUID3) String() string {
+	return string(u)
 }
 
 // UUID4 represents a uuid4 string format
@@ -467,6 +499,10 @@ func (u UUID4) Value() (driver.Value, error) {
 	return driver.Value(string(u)), nil
 }
 
+func (u UUID4) String() string {
+	return string(u)
+}
+
 // UUID5 represents a uuid5 string format
 //
 // swagger:strfmt uuid5
@@ -500,6 +536,10 @@ func (u *UUID5) Scan(raw interface{}) error {
 // Value converts a value to a database driver value
 func (u UUID5) Value() (driver.Value, error) {
 	return driver.Value(string(u)), nil
+}
+
+func (u UUID5) String() string {
+	return string(u)
 }
 
 // ISBN represents an isbn string format
@@ -537,6 +577,10 @@ func (u ISBN) Value() (driver.Value, error) {
 	return driver.Value(string(u)), nil
 }
 
+func (u ISBN) String() string {
+	return string(u)
+}
+
 // ISBN10 represents an isbn 10 string format
 //
 // swagger:strfmt isbn10
@@ -570,6 +614,10 @@ func (u *ISBN10) Scan(raw interface{}) error {
 // Value converts a value to a database driver value
 func (u ISBN10) Value() (driver.Value, error) {
 	return driver.Value(string(u)), nil
+}
+
+func (u ISBN10) String() string {
+	return string(u)
 }
 
 // ISBN13 represents an isbn 13 string format
@@ -607,6 +655,10 @@ func (u ISBN13) Value() (driver.Value, error) {
 	return driver.Value(string(u)), nil
 }
 
+func (u ISBN13) String() string {
+	return string(u)
+}
+
 // CreditCard represents a credit card string format
 //
 // swagger:strfmt creditcard
@@ -640,6 +692,10 @@ func (u *CreditCard) Scan(raw interface{}) error {
 // Value converts a value to a database driver value
 func (u CreditCard) Value() (driver.Value, error) {
 	return driver.Value(string(u)), nil
+}
+
+func (u CreditCard) String() string {
+	return string(u)
 }
 
 // SSN represents a social security string format
@@ -677,6 +733,10 @@ func (u SSN) Value() (driver.Value, error) {
 	return driver.Value(string(u)), nil
 }
 
+func (u SSN) String() string {
+	return string(u)
+}
+
 // HexColor represents a hex color string format
 //
 // swagger:strfmt hexcolor
@@ -710,6 +770,10 @@ func (h *HexColor) Scan(raw interface{}) error {
 // Value converts a value to a database driver value
 func (h HexColor) Value() (driver.Value, error) {
 	return driver.Value(string(h)), nil
+}
+
+func (h HexColor) String() string {
+	return string(h)
 }
 
 // RGBColor represents a RGB color string format
@@ -747,6 +811,10 @@ func (r RGBColor) Value() (driver.Value, error) {
 	return driver.Value(string(r)), nil
 }
 
+func (r RGBColor) String() string {
+	return string(r)
+}
+
 // Password represents a password.
 // This has no validations and is mainly used as a marker for UI components.
 //
@@ -781,4 +849,8 @@ func (r *Password) Scan(raw interface{}) error {
 // Value converts a value to a database driver value
 func (r Password) Value() (driver.Value, error) {
 	return driver.Value(string(r)), nil
+}
+
+func (r Password) String() string {
+	return string(r)
 }

--- a/strfmt/duration.go
+++ b/strfmt/duration.go
@@ -136,3 +136,8 @@ func (d *Duration) Scan(raw interface{}) error {
 func (d Duration) Value() (driver.Value, error) {
 	return driver.Value(int64(d)), nil
 }
+
+// String converts this duration to a string
+func (d Duration) String() string {
+	return time.Duration(d).String()
+}

--- a/strfmt/format.go
+++ b/strfmt/format.go
@@ -31,6 +31,7 @@ type Validator func(string) bool
 
 // Format represents a string format
 type Format interface {
+	String() string
 	encoding.TextMarshaler
 	encoding.TextUnmarshaler
 }

--- a/strfmt/format_test.go
+++ b/strfmt/format_test.go
@@ -32,6 +32,10 @@ func (t *testFormat) UnmarshalText(b []byte) error {
 	return nil
 }
 
+func (t testFormat) String() string {
+	return string(t)
+}
+
 func isTestFormat(s string) bool {
 	return strings.HasPrefix(s, "tf")
 }
@@ -51,6 +55,10 @@ func istf2(s string) bool {
 	return strings.HasPrefix(s, "af")
 }
 
+func (t tf2) String() string {
+	return string(t)
+}
+
 type bf string
 
 func (t bf) MarshalText() ([]byte, error) {
@@ -60,6 +68,10 @@ func (t bf) MarshalText() ([]byte, error) {
 func (t *bf) UnmarshalText(b []byte) error {
 	*t = bf(string(b))
 	return nil
+}
+
+func (t bf) String() string {
+	return string(t)
 }
 
 func isbf(s string) bool {

